crates/hyperswitch_domain_models/src/bulk_tokenization.rs

```diff
@@ -0,0 +1,308 @@
+    pub billing_address_line2: Option<masking::Secret<String>>,
+    pub billing_address_line3: Option<masking::Secret<String>>,
+    pub billing_address_zip: Option<masking::Secret<String>>,
+    pub billing_address_state: Option<masking::Secret<String>>,
+    pub billing_address_first_name: Option<masking::Secret<String>>,
+    pub billing_address_last_name: Option<masking::Secret<String>>,
+    pub billing_phone_number: Option<masking::Secret<String>>,
+    pub billing_phone_country_code: Option<String>,
+    pub billing_email: Option<pii::Email>,
+
+    // Other details
+    pub line_number: Option<u64>,
+    pub merchant_id: Option<id_type::MerchantId>,
+}
+
+impl ForeignFrom<&CardNetworkTokenizeRecord> for payments_api::CustomerDetails {
+    fn foreign_from(record: &CardNetworkTokenizeRecord) -> Self {
+        Self {
+            id: record.customer_id.clone(),
+            name: record.customer_name.clone(),
+            email: record.customer_email.clone(),
+            phone: record.customer_phone.clone(),
+            phone_country_code: record.customer_phone_country_code.clone(),
+        }
+    }
+}
+
+impl ForeignFrom<&CardNetworkTokenizeRecord> for payments_api::Address {
+    fn foreign_from(record: &CardNetworkTokenizeRecord) -> Self {
+        Self {
+            address: Some(payments_api::AddressDetails {
+                first_name: record.billing_address_first_name.clone(),
+                last_name: record.billing_address_last_name.clone(),
+                line1: record.billing_address_line1.clone(),
+                line2: record.billing_address_line2.clone(),
+                line3: record.billing_address_line3.clone(),
+                city: record.billing_address_city.clone(),
+                zip: record.billing_address_zip.clone(),
+                state: record.billing_address_state.clone(),
+                country: record.billing_address_country,
+            }),
+            phone: Some(payments_api::PhoneDetails {
+                number: record.billing_phone_number.clone(),
+                country_code: record.billing_phone_country_code.clone(),
+            }),
+            email: record.billing_email.clone(),
+        }
+    }
+}
+
+impl ForeignTryFrom<CardNetworkTokenizeRecord> for payment_methods_api::CardNetworkTokenizeRequest {
+    type Error = error_stack::Report<errors::ValidationError>;
+    fn foreign_try_from(record: CardNetworkTokenizeRecord) -> Result<Self, Self::Error> {
+        let billing = Some(payments_api::Address::foreign_from(&record));
+        let customer = payments_api::CustomerDetails::foreign_from(&record);
+        let merchant_id = record.merchant_id.get_required_value("merchant_id")?;
+
+        match (
+            record.raw_card_number,
+            record.card_expiry_month,
+            record.card_expiry_year,
+            record.payment_method_id,
+        ) {
+            (Some(raw_card_number), Some(card_expiry_month), Some(card_expiry_year), _) => {
```

Comment:
```suggestion
            (Some(raw_card_number), Some(card_expiry_month), Some(card_expiry_year), None) => {
```
we can have strict decision taken here. instead of `_` we can have None.
If both card and payment method id is present, we can throw an err